src/Domain/DomainHelpers.hpp

```diff
@@ -29,9 +31,16 @@ void set_internal_boundaries(
 /// stored in identifications.
 template <size_t VolumeDim>
 void set_periodic_boundaries(
-    const std::vector<std::vector<size_t>>& identifications,
+    const std::vector<pair_of_sets>& identifications,
     const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
         corners_of_all_blocks,
     gsl::not_null<std::vector<
         std::unordered_map<Direction<VolumeDim>, BlockNeighbor<VolumeDim>>>*>
         neighbors_of_all_blocks);
+
+/// struct for storing pairs of sets of corners in the global corner
+/// numbering scheme.
```

Comment:
I suggest this comment either point towards `Domain.hpp` where the purpose of the struct is described, or that the comment be changed to include a brief description of what is going on (e.g. what even is a "set of corners", why might I want to pair up two of them).